### ISSUE TITLE
Attribution Required by MIT License

### ISSUE THREAD

**Issue by NeuroAssassin** [mrTownsend/mrTownsend-Cogs#5]
Hello, I hope this finds you well.

I have noticed that several pieces of code that are licensed to me (available at https://github.com/NeuroAssassin/Toxic-Cogs) have been copied to this repository and have had proper credit removed from them.  According to the MIT license,
> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Several of the folders (namely `maintenance`, `scanner` (renamed to `imagescanning`) and `reacticket`) have been copied over, credit removed from the respective `info.json`'s and licensing information removed.  Under Github's DMCA policy, consider this as an attempt to negotiate prior to filing a DMCA violation with Github.

In your README file for your repository, you make the following statement:
> If any of you fellow devs see this and want a credit tag in the readme by all means ask. But as I move further and further away from your original code base, your credit will eventually be removed.

Due to the complexity of the code that has been copied over, these scripts do not constitute fair use and proper attribution is required.  Depending on the amount changed in the future, this may change, however at the time being credit is required.

I expect proper credit to be added within 24 hours of this notice.  If credit has been failed to assign, I will be contacting Github support to file a DMCA violation.  Thank you.

**Comment by mrTownsend** [comment]
Hello there, flaree! Let me clarify for all PR -

I'm still going to place in here credit for cogs from your repositories.

Only once a module has been *completely rewritten* will credits be removed. 

My method of grabbing all of these repo's just to put them in one place was a very hack-job solution, and many files including licenses were not brought along. Solely the code that makes it function, that is it. Over the next coming days you will start to see them appear as I source them from the original author, as well as credit tags on the main module repository page. I am true to my word, I just realized that I didn't have the mental capacity at the time of writing said readme to go back and find all of those files. It was....4AM, more than likely and I was probably higher than a kite.

If you have any further concerns please feel free to reply. I am not here trying to be a dickhead and rip other module owners. The RED community has a fantastic development base and there are many talented module owners and creators out there, and I'll be uploading my own originals as well. Many of the files have already had obvious changes to the authors and whatnot as I do prepare to make significant changes to them. You caught the painter before the paint was dry, I guess I could say.

Hope this clears any potential issues up. I'll be forwarding a copy of this to any collegues who reach out to me additionally. I am not here to create problems with you guys and once this is rectified I hope to continue to be an active contributor in the RED community.

//C

**Comment by NeuroAssassin** [comment]
> Only once a module has been completely rewritten will credits be removed.

I am glad to hear that, but please note that due to complexity of several of these cogs, they will need to be entirely rewritten and contain practically no copied code.  I personally have doubt that a similar cog can be created from reacticket (due to the sheer size) without copying code, but I'm not one to make assumptions so we'll have to wait and see.  Please keep in mind that it will still be at the discretion of me and Github support.

> My method of grabbing all of these repo's just to put them in one place was a very hack-job solution, and many files including licenses were not brought along.

While I can see that it may have been difficult to bring the LICSENSE file along, you still removed my name from the `info.json` files pertaining to my cogs, which is not a mistake that can be made easily while copying over.  You still deliberately removed my credit without changing the code a substantial amount to warrant re-crediting.

If you wished to start changing the credit name without the cog, then you should have started it locally or made this a private repository, where this would not have been an issue.

> If you have any further concerns please feel free to reply. I am not here trying to be a dickhead and rip other module owners.

While I understand this is likely your intention, it can be difficult to be a software developer and see your code taken from your repository, posted somewhere else and without your name attached.  It feels like we are being ripped off from our work.

**Comment by mrTownsend** [comment]
Hi! Thank you for actually not being a cunt replying to me. Better than 90% of my DM's right now, so thank you genuinely. 

I should additionally clarify that cogs that never make it to a fully rewritten state will never have the license file nor the credit link on the readme removed **intentionally**. In this case I'm getting my asshole ripped apart because I cloned folders containing code and not the folders that contained the folders that contained the code and packaged files.  

The info's of every file were changed in preparation for continuing rewrites and updates - this is not intended as a stick up the ass to the original creators. You'll find many version numbers were reset too to a "0.0.1" as i adopt them for my own update logging. If you happen to find any I haven't changed let me know as that'll piss my OCD off. 

I'm genuinely not here trying to be a menace and I've got overenthusiastic coders all in my mentions because some files were missing like I get the point, you know, credit is important, but it was genuinely a technical fuckup and not an intentional rip-off and it says so now in the readme, so I'm assuming anybody who continues to come at me after that update is just not reading it whatsoever. As issues are filed on the GitHub, I'm publicly actively taking accountability and responding instead of just making it private, or choosing to block the original developers, or take other actions to circumvent detection.

**Comment by mrTownsend** [comment]
LICENSES for imagescanning, reactickets, and maintenance have been added. Please let me know if there is anything I can do for you by opening a new issue. Thank you!

**Comment by NeuroAssassin** [comment]
I see that it was added, thank you very much.  Is there a way that my name can be added to the `author` field in the `info.json` files?  That should be all, and thanks for working through the process.

**Comment by mrTownsend** [comment]
The reacticket `info.json` was unchanged. It has not been modified. 

Credit has been added for `imagescanning` and `maintenance`. I apologize for the issues here. Please reach out and open a new issue if there's anything else I can ever do for you :)

**Comment by NeuroAssassin** [comment]
Great, thank you.
